nvflare/private/fed/server/job_runner.py

```diff
@@ -430,17 +430,20 @@ def restore_running_job(self, run_number: str, job_id: str, job_clients, snapsho
     def stop_run(self, job_id: str, fl_ctx: FLContext):
         engine = fl_ctx.get_engine()
         job_manager = engine.get_component(SystemComponents.JOB_MANAGER)
-        with self.lock:
-            self._stop_run(job_id, fl_ctx)
-            job = self.running_jobs.get(job_id)
-            if job:
-                self.log_info(fl_ctx, f"Stop the job run: {job_id}")
-                fl_ctx.set_prop(FLContextKey.CURRENT_JOB_ID, job.job_id)
-                job_manager.set_status(job.job_id, RunStatus.FINISHED_ABORTED, fl_ctx)
+        self._stop_run(job_id, fl_ctx)
+
+        # with self.lock:
```

Comment:
if we change this lock location, do we need to make sure the job_manager will function correctly if multiple threads calling it?

Reply:
Yes. The purpose of the lock is only to protect the self.running_jobs change. Reduced the scope to only necessary.